typeid/typeid-go/constructors.go

```diff
@@ -98,22 +98,39 @@ func split(id string) (string, string, error) {
 	}
 }
 
-// FromUUID encodes the given UUID (in hex string form) as a TypeID with the given prefix.
-func FromUUID[T Subtype, PT SubtypePtr[T]](prefix string, uidStr string) (T, error) {
+// FromUUID encodes the given UUID (in hex string form) as a TypeID
+func FromUUID[T Subtype, PT SubtypePtr[T]](uidStr string) (T, error) {
 	uid, err := uuid.FromString(uidStr)
 	var nilID T
 
 	if err != nil {
 		return nilID, err
 	}
 	suffix := base32.Encode(uid)
-	return from[T, PT](prefix, suffix)
+	return from[T, PT](nilID.Prefix(), suffix)
+}
+
+// FromUUIDBytes encodes the given UUID (in byte form) as a TypeID
+func FromUUIDBytes[T Subtype, PT SubtypePtr[T]](bytes []byte) (T, error) {
+	uidStr := uuid.FromBytesOrNil(bytes).String()
+	return FromUUID[T, PT](uidStr)
+}
+
+// AnyPrefixFromUUID encodes the given UUID (in hex string form) as a TypeID with the given prefix.
+func AnyPrefixFromUUID(prefix string, uidStr string) (AnyID, error) {
+	uid, err := uuid.FromString(uidStr)
```

Comment:
nit, but if you keep around a private implementation the takes in both the generic and the prefix, you can use it as the single implementation for both the AnyPrefix case and the typed case:
```
func fromUUID[T Subtype, PT SubtypePtr[T]](prefix string, uidStr string) (T, error) {
  ... // implementation here
 }
 
 func FromUUID [T Subtype, PT SubtypePtr[T]](uidStr string) (T, error) {
   // throw error if dealing with Any, see what we do in New()
   prefix := defaultPrefix[T]()
   return fromUUID[T, PT](prefix, uidstr)
  }
  
  func FromUUIDWithPrefix(prefix string, uidStr string) (AnyID, error) {
    return fromUUID[AnyID](prefix, uidStr)
   }
```

Same for the UUIDBytes implementation